inference-engine/src/mkldnn_plugin/nodes/mkldnn_roll_node.cpp

```diff
@@ -0,0 +1,214 @@
+    if (shiftTensorRank > 1) {
+        IE_THROW() << layerErrorPrefix << " doesn't support 'shift' input tensor with rank: " << shiftTensorRank;
+    }
+
+    shape = dataShape;
+}
+void MKLDNNRollNode::getSupportedDescriptors() {}
+
+void MKLDNNRollNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    auto inputData = getCnnLayer()->insData[0].lock();
+
+    if (inputData == nullptr) {
+        IE_THROW() << "Roll layer with name '" + getCnnLayer()->name + "'" << " has nullable data";
+    }
+
+    InferenceEngine::Precision precision = inputData->getPrecision();
+
+    auto dataType = MKLDNNExtensionUtils::IEPrecisionToDataType(precision);
+
+    auto srcDims = getParentEdgeAt(0)->getDims();
+
+    auto dataMemoryFormat = MKLDNNMemory::GetPlainFormat(getParentEdgeAt(0)->getDims());
+    InferenceEngine::LayerConfig config;
+    config.dynBatchSupport = true;
+
+    auto createDataConfig = [](const MKLDNNDims& dims, memory::data_type dataType) -> InferenceEngine::DataConfig {
+        InferenceEngine::DataConfig dataConfig;
+        dataConfig.inPlace = -1;
+        dataConfig.constant = false;
+        dataConfig.desc = MKLDNNMemoryDesc(dims, dataType, MKLDNNMemory::GetPlainFormat(dims));
+        return dataConfig;
+    };
+
+    config.inConfs.push_back(createDataConfig(getParentEdgeAt(0)->getDims(), dataType));
+    config.inConfs.push_back(createDataConfig(getParentEdgeAt(1)->getDims(), memory::data_type::s32));
+    config.inConfs.push_back(createDataConfig(getParentEdgeAt(2)->getDims(), memory::data_type::s32));
+
+    config.outConfs.push_back(createDataConfig(getChildEdgeAt(0)->getDims(), dataType));
+
+    supportedPrimitiveDescriptors.push_back({config, impl_desc_type::ref, dataMemoryFormat});
+}
+
+
+void MKLDNNRollNode::execute(mkldnn::stream strm) {
+    auto input = getParentEdgeAt(DATA_INDEX)->getBlob();
+    auto shifts = getParentEdgeAt(SHIFT_INDEX)->getBlob();
+    auto axes = getParentEdgeAt(AXES_INDEX)->getBlob();
+    auto output = getChildEdgeAt(0)->getBlob();
+    const auto dataPrecision = getInputPrecisions()[0];
+    const auto& dataTypeSize = dataPrecision.size();
+    switch (dataTypeSize) {
+        case sizeof(PrecisionTrait<Precision::I8>::value_type): {
+            rollImpl<PrecisionTrait<Precision::I8>::value_type>(input, shifts, axes, output);
+            break;
+        }
+        case sizeof(PrecisionTrait<Precision::I16>::value_type): {
+            rollImpl<PrecisionTrait<Precision::I16>::value_type>(input, shifts, axes, output);
+            break;
+        }
+        case sizeof(PrecisionTrait<Precision::I32>::value_type): {
+            rollImpl<PrecisionTrait<Precision::I32>::value_type>(input, shifts, axes, output);
+            break;
+        }
+        case sizeof(PrecisionTrait<Precision::I64>::value_type): {
+            rollImpl<PrecisionTrait<Precision::I64>::value_type>(input, shifts, axes, output);
```

Comment:
CPU plugin doesn't support 8 byte in general so I would suggest to remove the instance at all. It will help us to save binary size.